tests/unit/TextProcessorSystem/TextProcessorSystemTests/TextAnalyzerTests.cs

```diff
@@ -0,0 +1,31 @@
+﻿using TextProcessorSystem;
+
+namespace TextProcessorSystemTests
+{
+    public class TextAnalyzerTests
+    {
+        TextAnalyzer? textAnalyzer;
+
+        public TextAnalyzerTests()
+        {
+            this.textAnalyzer = new ();
+        }
+        [Fact]
+        public void AnalyzeText_WithPost_ShouldReturnDictionaryWithWordsAndFrequencies()
+        {
+            // Arrange
+            string post = "This \nshould be replaced . This \nshould not be replaced";
+            Dictionary<string, int> expectedFrequencyDictionary = new();
```

Comment:
```suggestion
            Dictionary<string, int> expectedFrequencyDictionary = new()
                                                       {
                                                           { "This", 2 },
                                                           { "", 2},
                                                       };
```

Instead of adding we can have like this

Reply:
Modified way of inserting in dictionary.